GiEnJul/ClientApp/src/components/admin/editLandingPage/EditMacro.tsx

```diff
@@ -29,74 +30,67 @@ export default () => {
                 alignItems="center">
-                <Grid item xs={2}>
-                    <Button variant='outlined' fullWidth={true} onClick={() => {
-                        setStep(1)
-                    }}>HVordan</Button>
-                </Grid>
-                <Grid item xs={2} >
-                    <Button variant='outlined' fullWidth={true} onClick={() => {
-                        setStep(2)
-                    }}>Spørsmål</Button>
-                </Grid>
-                <Grid item xs={2}>
-                    <Button variant='outlined' fullWidth={true} onClick={() => {
-                        setStep(3)
-                    }}>Bedrifter</Button>
-                </Grid>
-                <Grid item xs={2}>
-                    <Button variant='outlined' fullWidth={true} onClick={() => {
-                        setStep(4)
-                    }}>Kontakt</Button>
-                </Grid>
-                <Grid item xs={2}>
-                    <Button variant='outlined' fullWidth={true} onClick={() => {
-                        setStep(5)
-                    }}>Start "Gi en jul"</Button>
-                </Grid>
+                {sections.map((key, index) =>
+                    <Grid item xs={2}>
+                        <Button variant='outlined' fullWidth={true} onClick={() => {
+                            setStep(index)
+                        }}>{key}</Button>
+                    </Grid>)}
             </Grid>
-                    
-                    <EditHow
-                    step = {step}
-                    howState = {howState}
-                    setHowState = {setHowState}
-                    howTitleState = {howTitleState}
-                    setHowTitleState ={setHowTitleState}/>
-                    <EditQuestions
-                    step = {step}
-                    questionState ={questionState}
-                    setQuestionState ={setQuestionState}
-                    questionTitleState = {questionTitleState}
-                    setQuestionTitleState ={setQuestionTitleState}/>
-                    <EditCompany
-                    step = {step}
-                    companyState ={companyState}
-                    setCompanyState ={setCompanyState}
-                    companyTitleState = {companyTitleState}
-                    setCompanyTitleState ={setCompanyTitleState}/>
-                    <EditContact
-                    step = {step}
-                    contactState ={contactState}
-                    setContactState ={setContactState}
-                    contactTitleState = {contactTitleState}
-                    setContactTitleState ={setContactTitleState}/>
-                    <EditStart
-                    step = {step}
-                    startState ={startState}
-                    setStartState ={setStartState}
-                    startTitleState = {startTitleState}
-                    setStartTitleState ={setStartTitleState}/>
-
+            {(step === 0 &&
```

Comment:
I think this should start with 1 instead of 0. This applies to the next condition here as well (step === 1)

Reply:
the setStep function is set by the index of the sections array, and therefore my opinion is that it is natural to start at 0. If it starts at 1 the setStep would be setStep(index+1).

Reply:
Okay, I agree that this is more natural 👍